src/libcommon/log/sentry/sentryhandler.h

```diff
@@ -0,0 +1,105 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <sentry.h>
+#include <string>
+#include <mutex>
+
+#include "utility/types.h"
+#include "sentryuser.h"
+
+namespace KDC {
+
+enum class SentryLevel {  // Not defined in types.h as we don't want to include sentry.h in types.h
+    Debug = SENTRY_LEVEL_DEBUG,
+    Info = SENTRY_LEVEL_INFO,
+    Warning = SENTRY_LEVEL_WARNING,
+    Error = SENTRY_LEVEL_ERROR,
+    Fatal = SENTRY_LEVEL_FATAL
+};
+
+inline std::string toString(SentryLevel level) {
+    switch (level) {
+        case SentryLevel::Debug:
+            return "Debug";
+        case SentryLevel::Info:
+            return "Info";
+        case SentryLevel::Warning:
+            return "Warning";
+        case SentryLevel::Error:
+            return "Error";
+        case SentryLevel::Fatal:
+            return "Fatal";
+    }
+};
+
+class SentryHandler {
+    public:
+        enum class SentryProject { Server, Client, Disable };  // Only used for initialization, don't need to be in types.h
+
+        ~SentryHandler();
+        static std::shared_ptr<SentryHandler> instance();
+        static void init(SentryProject project, int breadCrumbsSize = 100);
+        void setAuthenticatedUser(const SentryUser &user);
+        void setGlobalConfidentialityLevel(SentryConfidentialityLevel level);
+        void captureMessage(SentryLevel level, const std::string &title, std::string message,
+                            const SentryUser &user = SentryUser());
+
+    private:
+        static std::shared_ptr<SentryHandler> _instance;
+        SentryHandler() = default;
+        SentryHandler(const SentryHandler &) = delete;
+        SentryHandler &operator=(const SentryHandler &) = delete;
+        SentryHandler(SentryHandler &&) = delete;
+        SentryHandler &operator=(SentryHandler &&) = delete;
+
+        sentry_value_t toSentryValue(const SentryUser &user);
+
+        struct SentryEvent {
+                using time_point = std::chrono::system_clock::time_point;
+
+                SentryEvent(const std::string &title, const std::string &message, SentryLevel level,
+                            SentryConfidentialityLevel userType, const SentryUser &user);
+                std::string getStr() const { return title + message + (char)toInt(level) + userId; };
+                std::string title;
+                std::string message;
+                SentryLevel level;
+                SentryConfidentialityLevel confidentialityLevel = SentryConfidentialityLevel::Anonymous;
+                std::string userId;
+                time_point lastCapture;
+                time_point lastUpload;
+                unsigned int captureCount = 0;
+        };
+
+        std::mutex _mutex;
+        bool _isSentryActivated = false;
```

Comment:
Depending on the final choice of the enum name `SentryProject` above, you can rename this variable as `_isSentryEnabled`.